Release 5.6.0 (#2015)
microsoft/onefuzz: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## 5.6.0
### Added

> * Service: Add Function App Settings to Bicep Template and deploy.py. [#1973](https://github.com/microsoft/onefuzz/pull/1973)

Reviewer: ```suggestion
* Service: Add Function App settings to Bicep template and `deploy.py`. [#1973](https://github.com/microsoft/onefuzz/pull/1973)
```

> * Agent: Add a timestamp to Agent log to make it easier to correlate events. [#1972](https://github.com/microsoft/onefuzz/pull/1972)

Reviewer: ```suggestion
* Agent: Add a timestamp to agent log to make it easier to correlate events. [#1972](https://github.com/microsoft/onefuzz/pull/1972)
```

### Changed

> * Agent/Supervisor/Proxy: Rename the supervisor process from `onefuzz-supervisor` to `onefuzz-agent`. [#1989](https://github.com/microsoft/onefuzz/pull/1989)

Reviewer: Note for future changelogs: we longer have a "supervisor". We should probably start referring to the new `onefuzz-agent` as "the agent" and `onefuzz-task` as the task worker / executor.

Author: Shall I make the edit for this round? or wait until next release?

Reviewer: Next is fine! 👍

> * Agent/Supervisor/Proxy: Rename the agent process that runs on the VM from `onefuzz-agent` to `onefuzz_task`. [#1980](https://github.com/microsoft/onefuzz/pull/1980)

Reviewer: ```suggestion
* Agent/Supervisor/Proxy: Rename the task executor that runs on the VM from `onefuzz-agent` to `onefuzz-task`. [#1980](https://github.com/microsoft/onefuzz/pull/1980)
```

> * Agent/Supervisor/Proxy: Ensure `GlobalFlag` Reg Value is Initialized for Target. [#1960](https://github.com/microsoft/onefuzz/pull/1960)

Reviewer: ```suggestion
* Agent/Supervisor/Proxy: Ensure `GlobalFlag` registry value is initialized for targets. [#1960](https://github.com/microsoft/onefuzz/pull/1960)
```

> * Agent/Supervisor/Proxy: Set Backtrace to Full. [#1959](https://github.com/microsoft/onefuzz/pull/1959)

Reviewer: ```suggestion
* Agent/Supervisor/Proxy: Enable full backtraces on Rust panics. [#1959](https://github.com/microsoft/onefuzz/pull/1959)
```

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## 5.6.0
### Added
* Service: Add Function App Settings to Bicep Template and deploy.py. [#1973](https://github.com/microsoft/onefuzz/pull/1973)

> * Agent: Add Timestamp to Agent Log. [#1972](https://github.com/microsoft/onefuzz/pull/1972)

Reviewer 2: ```suggestion
* Agent: Add a timestamp to Agent log to make it easier to correlate events. [#1972](https://github.com/microsoft/onefuzz/pull/1972)
```

### Changed

> * Agent/Supervisor/Proxy: Rename Supervisor to Agent. [#1989](https://github.com/microsoft/onefuzz/pull/1989)

Reviewer 2: ```suggestion
* Agent/Supervisor/Proxy: Rename the supervisor process from `onefuzz-supervisor` to `onefuzz-agent`. [#1989](https://github.com/microsoft/onefuzz/pull/1989)
```

> * Agent/Supervisor/Proxy: Rename Agent to Task. [#1980](https://github.com/microsoft/onefuzz/pull/1980)

Reviewer 2: ```suggestion
* Agent/Supervisor/Proxy: Rename the agent process that runs on the VM from `onefuzz-agent` to `onefuzz_task`. [#1980](https://github.com/microsoft/onefuzz/pull/1980)
```

* Agent/Supervisor/Proxy: Ensure `GlobalFlag` Reg Value is Initialized for Target. [#1960](https://github.com/microsoft/onefuzz/pull/1960)
* Agent/Supervisor/Proxy: Set Backtrace to Full. [#1959](https://github.com/microsoft/onefuzz/pull/1959)

> * Agent/Supervisor/Proxy: Updated multiple third-party Rust dependencies. [#2003](https://github.com/microsoft/onefuzz/pull/2003), [#2002](https://github.com/microsoft/onefuzz/pull/2002), [#1999](https://github.com/microsoft/onefuzz/pull/1999), [#1992](https://github.com/microsoft/onefuzz/pull/1992), [#1986](https://github.com/microsoft/onefuzz/pull/1986), [#1983](https://github.com/microsoft/onefuzz/pull/1983), [#1982](https://github.com/microsoft/onefuzz/pull/1982), [#1981](https://github.com/microsoft/onefuzz/pull/1981), [#1985](https://github.com/microsoft/onefuzz/pull/1985), [#1974](https://github.com/microsoft/onefuzz/pull/1974), [#1969](https://github.com/microsoft/onefuzz/pull/1969), [#1965](https://github.com/microsoft/onefuzz/pull/1965), 

Reviewer 2: ```suggestion
* Agent/Supervisor/Proxy: Updated multiple third-party Rust dependencies. [#2003](https://github.com/microsoft/onefuzz/pull/2003), [#2002](https://github.com/microsoft/onefuzz/pull/2002), [#1999](https://github.com/microsoft/onefuzz/pull/1999), [#1992](https://github.com/microsoft/onefuzz/pull/1992), [#1986](https://github.com/microsoft/onefuzz/pull/1986), [#1983](https://github.com/microsoft/onefuzz/pull/1983), [#1982](https://github.com/microsoft/onefuzz/pull/1982), [#1981](https://github.com/microsoft/onefuzz/pull/1981), [#1985](https://github.com/microsoft/onefuzz/pull/1985), [#1974](https://github.com/microsoft/onefuzz/pull/1974), [#1969](https://github.com/microsoft/onefuzz/pull/1969), [#1965](https://github.com/microsoft/onefuzz/pull/1965)
```

* CLI/Service: Updated multiple first-party and third-party Python dependencies. [#2009](https://github.com/microsoft/onefuzz/pull/2009), [#1996](https://github.com/microsoft/onefuzz/pull/1996)

### Fixed 

> * Agent: Removing Stray Print from task_logger. [#1975](https://github.com/microsoft/onefuzz/pull/1975)

Reviewer 2: ```suggestion
* Agent: Remove stray print statement from the task_logger. [#1975](https://github.com/microsoft/onefuzz/pull/1975)
```

> * Agent: Fix Local Coverage Definition. [#1962](https://github.com/microsoft/onefuzz/pull/1962)

Reviewer 2: ```suggestion
* Agent: Fix local coverage definition by removing a duplicated command line parameter. [#1962](https://github.com/microsoft/onefuzz/pull/1962)
```

> * Build: Fix YML Dependabot Formatting. [#1998](https://github.com/microsoft/onefuzz/pull/1998)

Reviewer 2: ```suggestion
```

> * Service: Bug Fix in TimerTasks Function. [#1961](https://github.com/microsoft/onefuzz/pull/1961)

Reviewer 2: ```suggestion
```
